### ISSUE TITLE
[Spells][Shaman] Grounding totem not destroyed

### ISSUE THREAD

**Issue by devastator** [FeenixServerProject/Archangel_2.4.3_Bugtracker#2536]
Current State: Grounding totem is not destroyed upon using certain spells. It redirects those spells to itself, and has no effect after that (it is effectively destroyed) but the Grounding Totem itself is not destroyed and persists as animation which causes distraction.

Expected Result: Grounding totem should get destroyed upon redirecting any spell on itself.
From the 2.3 patch notes:
Grounding Totem: This totem is now destroyed upon redirecting any spell to itself.

The spells that cause the issue are spells that cannot directly "kill" the totem as the direct damage spells. So far found the following:
Non-damaging (effect) spells: Warlock Curses, Priest Vampiric Embrace, Dispel Magic, Silence, Mage Counterspell etc.
Channeled spells: Warlock Drain Life, Drain Soul, Drain Mana, Priest Mind Flay etc
DoTs: Warlock Curse of Agony, Corruption, Siphon Life, Priest Vampiric Touch, Shadow Word: Pain, Devouring Plague, Starshards
Traps: Hunter Immolation and Freezing Trap https://github.com/FeenixServerProject/Archangel_2.4.3_Bugtracker/issues/2136

Reproduction Guide: You will need 2 clients
Client 1:
Create Draenei Shaman
.level 69
.learn 8177
.t sw
Client 2:
Create Horde Priest
.level 69
.learn all_myspells
.t sw

Put grounding totem with the shaman, then use any of the mentioned Priest abilities on the Shaman - notice that they will get redirected to the totem, the totem will lose its power (for most spells it will say Immune) - it won't redirect any other spell to itself afterwards, but it will stay as object not being destroyed until new grounding totem is put.

**Comment by griesakanik** [comment]
is this still happening ?

**Comment by Rakanish** [comment]
Yes it still happens, for instance using faerie fire to take out grounding will not destroy it visually.

**Comment by asdsha** [comment]
Grounding totems stays up if redirected spell causes no damage, quite annoying.

**Comment by TOSMANOTA** [comment]
up, no changelogs for G totem ?

**Comment by devidao** [comment]
Fixed, coming soon on PTR
